R/dataset.R

```diff
@@ -329,12 +329,39 @@ setMethod("allVariables<-", c("CrunchDataset", "VariableCatalog"),
 setMethod("hidden", "CrunchDataset", function (x) hidden(allVariables(x)))
 
 
-webURL <- function (x) {
+APIToWebURL <- function (x) {
     ## URL to view this dataset in the web app
     stopifnot(is.dataset(x))
     return(paste0(absoluteURL("/", getOption("crunch.api")), "dataset/", id(x)))
 }
 
+webToAPIURL <- function (url) {
+    id <- sub("^https.*?/dataset/([0-9a-f]+)/.*$", "\\1", url)
```

Comment:
Ah, yeah I think having `datasetReference` match either won't break anything, and it will keep our parsing all in one place. There's a bug I fixed in the conditional branch (https://github.com/Crunch-io/rcrunch/pull/65/files#diff-3647057bf9eebd954d358f8bc6e4beedR168) that bit Brian using logical expressions. 

It's not a big deal if we have them separate, I just remembered running into a problem parsing urls for ids earlier.